packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go

```diff
@@ -167,6 +175,26 @@ func New(opts ...Option) (*DHTPeer, error) {
 		return nil, errors.New("public host and port must be set")
 	}
 
+	// check if the PoR is delivered for my public  key
+	if dhtPeer.myAgentRecord != nil {
```

Comment:
[STP2] peer will verify agent PoR before registering it the agent and saves it locally